### PR TITLE
WIP: Port to Qt 5.11

### PR DESCRIPTION
**Pull request by z3ntu**
I've made the theme compile and (at least partially - I don't know if the issues are because of the application I've tested it with, my changes or the theme itself) work under Qt 5.11.

### CODE REVIEW

**Comment by cibersheep** [comment]
@z3ntu any new on this ^__^ ?

**Comment by z3ntu** [comment]
@cibersheep Not much except that I've made adjustments for Qt 5.12 in https://github.com/z3ntu/qqc2-suru-style/commit/7fca387ff87a5f940b2aca065dd7d86e08ef33d5 and it seems to work fine.

**Comment by cibersheep** [comment]
Ouh. that sound thrilling! I'll take a look. 5.12 is lts, right?

**Comment by JBBgameich** [comment]
Yes, it is a lts release.

**Comment by bhdouglass** [comment]
I'm closing this PR since it's against the master branch and the master branch is going away. Feel free to make a new PR against the xenial branch.

**Comment by z3ntu** [comment]
It's against the xenial branch now.

**Comment by cibersheep** [comment]
We want LTS, we love @z3ntu

File: qqc2-suru/qquicksurustyle.cpp [hunk 1]
```diff
@@ -89,16 +90,14 @@ static uint globalColors[2][9] = {
 
 static bool globalCustoms[2][9] = { { false } };
 
-extern bool qt_is_dark_system_theme();
```
Comment (bhdouglass) [inline qqc2-suru/qquicksurustyle.cpp]: This function was implemented, can you make sure those changes get merged into this branch?

File: qqc2-suru/qquicksurustyle.cpp [hunk 2]
```diff
@@ -89,16 +90,14 @@ static uint globalColors[2][9] = {
...
-
 static QQuickSuruStyle::Theme qquicksuru_effective_theme(QQuickSuruStyle::Theme theme)
 {
     if (theme == QQuickSuruStyle::System)
-        theme = qt_is_dark_system_theme() ? QQuickSuruStyle::Dark : QQuickSuruStyle::Light;
+        theme = QQuickStylePrivate::isDarkSystemTheme() ? QQuickSuruStyle::Dark : QQuickSuruStyle::Light;
     return theme;
 }
 
-QQuickSuruStyle::QQuickSuruStyle(QObject *parent) : QQuickStyleAttached(parent),
```
Comment (bhdouglass) [inline qqc2-suru/qquicksurustyle.cpp]: Is it possible to write this in such a way with preprocessor directives that is compiles and works for both 5.9 and 5.11?

File: qqc2-suru/qquicksurustyle_p.h [hunk 3]
```diff
@@ -24,8 +24,10 @@
 #ifndef QQUICKSURUSTYLE_P_H
 #define QQUICKSURUSTYLE_P_H
 
+#include <QSettings>
 #include <QtGui/qcolor.h>
-#include <QtQuickControls2/private/qquickstyleattached_p.h>
+//#include <QtQuickControls2/private/qquickstyleattached_p.h>
```
Comment (bhdouglass) [inline qqc2-suru/qquicksurustyle_p.h]: Since we don't need these any more can you remove them?

File: qqc2-suru/qtquickcontrols2surustyleplugin.cpp [hunk 4]
```diff
@@ -62,14 +64,16 @@ void QtQuickControls2SuruStylePlugin::registerTypes(const char *uri)
 {
     qmlRegisterType<QQuickSuruAnimations>();
     qmlRegisterType<QQuickSuruUnits>();
+    QByteArray import = QByteArray(uri) + ".impl";
+    qmlRegisterType(typeUrl(QStringLiteral("HighlightFocusRectangle.qml")), import, 2, 2, "HighlightFocusRectangle");
```
Comment (bhdouglass) [inline qqc2-suru/qtquickcontrols2surustyleplugin.cpp]: Why is this necessary?

File: qqc2-suru/qtquickcontrols2surustyleplugin.cpp [hunk 5]
```diff
@@ -62,14 +64,16 @@ void QtQuickControls2SuruStylePlugin::registerTypes(const char *uri)
...
     qmlRegisterUncreatableType<QQuickSuruStyle>(uri, 2, 2, "Suru", tr("Suru is an attached property"));
 }
 
 void QtQuickControls2SuruStylePlugin::initializeEngine(QQmlEngine *engine, const char *uri)
 {
     QQuickStylePlugin::initializeEngine(engine, uri);
 
-    engine->addImageProvider(name(), new QQuickColorImageProvider(QStringLiteral(":/qt-project.org/imports/QtQuick/Controls.2/Suru/assets")));
+    //engine->addImageProvider(name(), new QQuickColorImageProvider(QStringLiteral(":/qt-project.org/imports/QtQuick/Controls.2/Suru/assets")));
```
Comment (bhdouglass) [inline qqc2-suru/qtquickcontrols2surustyleplugin.cpp]: Why did this change? Having the shortcut makes things a bit simpler.

Comment (z3ntu) [reply qqc2-suru/qtquickcontrols2surustyleplugin.cpp]: See https://github.com/qt/qtquickcontrols2/commit/9fc3659a678e6d1bd6b2514486bde1d1725963da

File: qqc2-suru/qtquickcontrols2surustyleplugin.cpp (hunk 4 above)
Comment (z3ntu) [reply qqc2-suru/qtquickcontrols2surustyleplugin.cpp]: Removed

File: qqc2-suru/qquicksurustyle_p.h (hunk 3 above)
Comment (z3ntu) [reply qqc2-suru/qquicksurustyle_p.h]: Removed

File: qqc2-suru/qquicksurustyle.cpp (hunk 1 above)
Comment (z3ntu) [reply qqc2-suru/qquicksurustyle.cpp]: Done

**Comment by z3ntu** [comment]
Closed on favor of #18, let's continue there.
